packages/launchpad/src/use-tracking/index.tsx

```diff
@@ -0,0 +1,71 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { SiteDetailsOptions } from '@automattic/data-stores';
+import { useEffect, useMemo } from 'react';
+import type { Task } from '../types';
+
+interface LogParams {
+	tasks: Task[];
+	checklistSlug: string | null;
+	context: string;
+	siteIntent: SiteDetailsOptions[ 'site_intent' ];
+	flow?: string;
+}
+
+export const useTracking = ( params: LogParams ) => {
+	const { tasks, checklistSlug, context, siteIntent, flow } = params;
+	const completedSteps = useMemo( () => tasks.filter( ( task ) => task.completed ), [ tasks ] );
+	const taskNames = useMemo( () => tasks.map( ( task ) => task.id ).join( ',' ), [ tasks ] );
+	const numberOfSteps = tasks.length;
+	const numberOfCompletedSteps = completedSteps.length;
+	const isCheckListCompleted = completedSteps.length === tasks.length;
+
+	useEffect( () => {
+		if ( tasks.length === 0 || ! siteIntent ) {
+			return;
+		}
+
+		recordTracksEvent( 'calypso_launchpad_tasklist_viewed', {
+			number_of_steps: numberOfSteps,
+			number_of_completed_steps: numberOfCompletedSteps,
+			is_completed: isCheckListCompleted,
+			tasks: taskNames,
+			checklist_slug: checklistSlug,
+			context: context,
+			site_intent: siteIntent,
+			flow,
```

Comment:
Question: why are we adding `flow` to the event props? The current code isn't using this for the `calypso_launchpad_tasklist_viewed` event. For context, this is only relevant in the fullscreen launchpad, but it's not something we're tracking now. And it could be somewhat inferred from `context`.

Reply:
I was planning to use it on the next PR after merging this one but I am ok to remove it, removed here https://github.com/Automattic/wp-calypso/pull/87141/commits/77535c0f270825f9113487859561b6317aee3a11